### PR TITLE
Constant definition update for issue 41677

### LINKED ISSUE #41677: "Constants in C# are best defined as static members"
### Type of issue

Other (describe below)

### Description

This article includes this Note:

> The #define directive cannot be used to declare constant values as is typically done in C and C++. Constants in C# are best defined as static members of a class or struct. If you have several such constants, consider creating a separate "Constants" class to hold them.

Shouldn't this say that constants in C# are best defined with the [`const`](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/const) keyword? Using static members has different implications for the application, since static members are resolved at runtime and are not necessarily read-only. For primitive values, `const` in C# is much closer to the behavior of `#define` in C/C++, since both are resolved at compile time.

### Page URL

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives

### Content source URL

https://github.com/dotnet/docs/blob/main/docs/csharp/language-reference/preprocessor-directives.md

### Document Version Independent Id

501be68d-5ad0-18d9-8958-54215bb57fc0

### Article author

@BillWagner

### Metadata

* ID: 716913e5-dbbe-bdb0-909a-4d8c2dc7be97 
* Service: **dotnet-csharp**
* Sub-service: **lang-reference**

### PR DESCRIPTION
**Pull request by samwherever**
## Summary

I rewrote the note, based on the comment from @kevinfoley in the linked issue. 

(hopefully) Fixes #41677 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/language-reference/preprocessor-directives.md](https://github.com/dotnet/docs/blob/82cbe58701055f97308d0b15b462712e72ea37b3/docs/csharp/language-reference/preprocessor-directives.md) | [C# preprocessor directives](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives?branch=pr-en-us-41726) |
| [docs/csharp/language-reference/xmldoc/recommended-tags.md](https://github.com/dotnet/docs/blob/82cbe58701055f97308d0b15b462712e72ea37b3/docs/csharp/language-reference/xmldoc/recommended-tags.md) | [Recommended XML tags for C# documentation comments](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags?branch=pr-en-us-41726) |

### CODE REVIEW

File: docs/csharp/language-reference/preprocessor-directives.md [hunk 1]
```diff
@@ -184,7 +184,7 @@ You use `#define` to define a symbol. When you use the symbol as the expression
  ```
 
 > [!NOTE]
-> The `#define` directive cannot be used to declare constant values as is typically done in C and C++. Constants in C# are best defined as static members of a class or struct. If you have several such constants, consider creating a separate "Constants" class to hold them.
+> In C#, primitive constants should be defined using the `const` keyword. The `#define` directive cannot be used to declare constant values as is typically done in C and C++. You can also define constants in C# as static members of a class or struct, but be aware that static members are resolved at runtime, and thus not necessarily read-only. If you have several such constants, consider creating a separate "Constants" class to hold them.
```
Comment (BillWagner) [inline docs/csharp/language-reference/preprocessor-directives.md:187]: The change here could be misinterpreted by some readers. Members declared with the `const` keyword are `static` members. The distinction between using the `const` keyword and the `static` keyword is that `const` members are readonly. How about this wording instead:

```suggestion
> In C#, primitive constants should be defined using the [`const`](keywords/const.md) keyword. A `const` declaration creates a `static` member that can't be modified at runtime. The `#define` directive can't be used to declare constant values as is typically done in C and C++. If you have several such constants, consider creating a separate "Constants" class to hold them.
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks again @samwherever 

This is close, but I have some suggestions for different wording. What do you think?

**Comment by samwherever** [comment]
Thanks @BillWagner! I committed your changes. Thanks for the feedback!

**Review by BillWagner — APPROVED** [review]
